src/controllers/Interactions.ts

```diff
@@ -1,196 +1,125 @@
 /* eslint-disable prefer-const */
```

Comment:
Not disabling this rule would avoid me having to comment on every line that we unnecessarily use `let` instead of `const`. Const is preferred since it guarantees we can't reassign the variable to some other type which is good for data integrity and makes code easier to read/follow.